helm/prometheus-agent-app/templates/prometheus.yaml

```diff
@@ -0,0 +1,53 @@
+apiVersion: monitoring.coreos.com/v1
+kind: Prometheus
+metadata:
+  labels:
+     resource: Prometheus
+  name: {{ include "name" . }}
+  namespace: {{ .Release.Namespace }}
+spec:
+  containers:
+  - args:
+    - --config.file=/etc/prometheus/config_out/prometheus.env.yaml
+    - --storage.agent.path=/prometheus
+    - --enable-feature=agent
+    - --web.enable-lifecycle
+    - --web.listen-address=:9090
+    name: prometheus
+  enableAdminAPI: false
+  enableFeatures:
+  - agent
+  evaluationInterval: 30s
+  externalUrl: http://kube-prometheus-agent-prometheus.monitoring:9090
+  image: quay.io/giantswarm/prometheus:v2.36.1
+  listenLocal: false
+  logFormat: logfmt
+  logLevel: info
+  paused: false
+  podMonitorNamespaceSelector: {}
+  podMonitorSelector:
+    matchLabels:
+      release: kube-prometheus
+  portName: http-web
+  probeNamespaceSelector: {}
+  probeSelector:
+    matchLabels:
+      release: kube-prometheus
+  remoteWrite:
+  - url: http://thanos-receive-ingestor-default.thanos.svc:19291/api/v1/receive
+  replicas: 1
+  retention: 10d
+  routePrefix: /
+  scrapeInterval: 30s
+  securityContext:
+    fsGroup: 2000
+    runAsGroup: 2000
+    runAsNonRoot: true
+    runAsUser: 1000
+  serviceAccountName: kube-prometheus-agent-prometheus
+  serviceMonitorNamespaceSelector: {}
+  serviceMonitorSelector:
```

Comment:
serviceMonitorSelector should be set in values file as we will change this value